src/main/java/org/folio/rest/impl/BLUsersAPI.java

```diff
@@ -816,27 +829,57 @@ public void postBlUsersLogin(boolean expandPerms, List<String> include, String u
           .ifPresent(header -> headers.put(HttpHeaders.USER_AGENT, header));
         Optional.ofNullable(xForwardedFor)
           .ifPresent(header -> headers.put(X_FORWARDED_FOR_HEADER, header));
-        loginResponse[0] = client.request(HttpMethod.POST, entity, moduleURL, headers);
+
+        List<String> finalInclude = include;
+
+        clientForLogin.request(HttpMethod.POST, entity, moduleURL, headers)
+          .thenAccept(loginResponse -> {
+            //then get user by username, inject okapi headers from the login response into the user request
+            //see 'true' flag passed into the chainedRequest
+            handleResponse(loginResponse, false, false, true, aRequestHasFailed, asyncResultHandler);
+
+            String token = loginResponse.getHeaders().get(OKAPI_TOKEN_HEADER);
+            String tenant = getTenant(token);
+            okapiHeaders.put(OKAPI_TENANT_HEADER, tenant);
+            HttpClientInterface client = HttpClientFactory.getHttpClient(okapiURL, tenant);
+
+            try {
+              getUserWithPerms(expandPerms, okapiHeaders, asyncResultHandler, userUrl, finalInclude, token, tenant, client);
+            } catch (Exception e) {
+              client.closeClient();
+              asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(
```

Comment:
You can remove `io.vertx.core` here an in a few other places.

Reply:
Was only suggesting making the change on the new code, rather than the existing code which may have "broken" test coverage on sonar by putting new methods in scope for the change. I'll leave it up to you to decide how to handle that if at all.